react/OrderForm.tsx

```diff
@@ -117,9 +117,14 @@ export const OrderFormProvider: FC = ({ children }) => {
     const localOrderFormString = localStorage.getItem('orderform')
 
     if (localOrderFormString != null) {
-      const localOrderForm = JSON.parse(localOrderFormString)
-
-      if (localOrderForm.value !== UNSYNC_ORDER_FORM_VALUE) {
+      const localOrderForm = JSON.parse(localOrderFormString) as OrderForm
+
+      if (
+        localOrderForm.value !== UNSYNC_ORDER_FORM_VALUE &&
+        (!data ||
+          data.orderForm.id === localOrderForm.id ||
+          localOrderForm.id === DEFAULT_ORDER_FORM.id)
```

Comment:
When do `localOrderForm.value !== UNSYNC_ORDER_FORM_VALUE`  and `localOrderForm.id === DEFAULT_ORDER_FORM.id` happen at the same time?

Reply:
if you add an item while offline and it didn't synchronize with the server yet, but maybe in this case it doesn't make much sense.

Reply:
confirmed, it doesn't work if you add items when offline and go back online with a page refresh. it briefly removes all the items of the cart

Reply:
I was trying to write a test to cover this case but the logic for adding the item and creating the task to insert in the queue is in `order-items` :-(